cibuildwheel/macos.py

```diff
@@ -232,8 +231,29 @@ def setup_python(
         )
         sys.exit(1)
 
-    # install pip & wheel
-    call(["python", get_pip_script, *dependency_constraint_flags], env=env, cwd="/tmp")
+    # ensure pip is installed
+    if not (installation_bin_path / "pip").exists():
+        # perhaps pip is installed, but not available as 'pip'...
+        try:
+            call(["python", "-m", "pip", "--version"], env=env, cwd="/tmp")
+        except subprocess.CalledProcessError:
+            pip_is_installed = False
+        else:
+            pip_is_installed = True
```

Comment:
haha, yes that was my first instinct, too. You can see it here:
https://github.com/joerick/cibuildwheel/pull/665/commits/ef9163550e375127f25e6fa04a58a01b58e23f8e 

However, I ended up (https://github.com/joerick/cibuildwheel/pull/665#discussion_r628860733) changing that to the exception form because
- adding the `check` argument to `call()` is awkward, because it looks a lot like the `env` and `cwd` arguments, that passthrough to `subprocess.run`, but there's a subtle difference - a different default. 
- adding the return type CompletedProcess to `call()` is also awkward, because it's only useful when `check=False`. 
- it muddles `call()`'s purpose, it's no longer 'do this' it's now 'do this/try this and tell me what happened'

So I thought the exception was neater overall.

Reply:
Our entire wrapper is a bit bothersome - the important thing it does is add logging. It also happens to modify the interface just a tiny bit (different name, check=True). This is heavily because it was a shortcut for `check_call`, but we've standardized on `run(check=True)`, which is the recommended method to use for 3.5+ code. So what about renaming the wrapper to `utils.run`, just passing `**kwargs`, and adding `check=True` for most of the uses? I'd rather make it look like something people are used to (subprocess.run), rather than save a few chars.

Now if it was dramatically better, ala [plumbum](https://plumbum.readthedocs.io/en/latest/), then it would be worth learning something new. But not for just a tiny small change.

Could be updated later, of course, doesn't have to be in this PR.

Reply:
Points 2 and 3 are just inherited from `subprocess.run`. The interface might not be ideal, but it's in the stdlib.

Reply:
Would using:
```
    call(["python", "-m", "ensurepip"], env=env, cwd="/tmp")
    call(
         ["python", "-m", "pip", "install", "--force-reinstall", "pip", *dependency_constraint_flags],
         env=env,
         cwd="/tmp",
     )
```
solve the issue of detecting if pip is installed in the first place ?

Reply:
Would that ensure the defaultpip script was there? I think so?

Reply:
Given that an actual update would end-up in the place where we want to be and starts by removing the existing installation of pip, I would guess so. Let's try this. @joerick, I'll push a commit, please do revert it if you don't like what you're seeing :-)

Reply:
I agree- if that works, I'm interested. Let's see.